server/base/src/main/java/org/apache/accumulo/server/util/FileUtil.java

```diff
@@ -544,10 +543,10 @@ public static <T extends TabletFile> WritableComparable<Key> findLastKey(ServerC
           continue;
         }
 
-        Key key = reader.getLastKey();
+        Text row = reader.getLastRow();
 
-        if (lastKey == null || key.compareTo(lastKey) > 0) {
-          lastKey = key;
+        if (lastRow == null || row.compareTo(lastRow) > 0) {
```

Comment:
This is also an existing problem, not really handling fact that row could be null.  We could open a follow on issue to investigate.